### PR TITLE
Update test-network chaincode container versions

### LINKED ISSUE #1267: Cannot deploy Node or Java chaincode with Fabric v3
Deployment of Node (or Java) chaincode fails when using the test-network with Fabric v3.0. The following error occurs:

> Error: chaincode install failed with status: 500 - failed to invoke backing implementation of 'InstallChaincode': could not build chaincode: docker build failed: docker image build failed: docker build failed: Failed to pull hyperledger/fabric-nodeenv:3.0: API error (404): manifest for hyperledger/fabric-nodeenv:3.0 not found: manifest unknown: manifest unknown

This looks to be because the test-network peer configuration specifies `$(TWO_DIGIT_VERSION)` and the Node and Java chaincode container Docker image tag. While `3:0` tags do exist for the chaincode builder and Go chaincode images, Fabric v3 continues to use the `2.5` tagged Node and Java chaincode images.

### PR DESCRIPTION
**Pull request by bestbeforetoday**
The test-network peer configuration specifies $(TWO_DIGIT_VERSION) as the tag for the Node and Java chaincode containers. For Fabric v3.0, this requests fabric-nodeenv:3.0 and fabric-javaenv:3.0 Docker images to host Node and Java chaincode respectively. These images do not exist, which causes deployment of Node and Java chaincode to fail when using Fabric v3.0. Fabric v3.0 continues to use fabric-nodeenv:2.5 and fabric-javaenv:2.5.

This change updates the test-network peer configuration to explicitly specify `2.5` as the Node and Java chaincode Docker image tags. This is (currently) the correct version for both Fabric v2.5 and Fabric v3.0.

Closes #1267

### CODE REVIEW

**Comment by satota2** [comment]
@bestbeforetoday 

Thank you for the update.

I noticed that this update changes the indentation in the YAML files from 2 spaces to 4. Is there a specific reason for this change? Other files appear to be consistently using 2 spaces, so it might be better to align with that convention for consistency. What do you think?

**Comment by bestbeforetoday** [comment]
I think this PR changes the indentation from 4 spaces to the same 2 spaces used in other Yaml files.

**Comment by satota2** [comment]
@bestbeforetoday 
Oh, my apologies! I somehow misunderstood, even though you had already corrected it to 2 spaces. Please disregard my previous comment… I’m a bit embarrassed about the mix-up!

**Comment by bestbeforetoday** [comment]
No problem! I didn't set out to change the whitespace intentionally. I just have my editor set to autoformat on save. If you are happy with the 2 space indentation, the changes are (much) easier to review if you set GitHub's diff view settings to ignore whitespace.

File: test-network/addOrg3/compose/podman/peercfg/core.yaml
Comment (satota2) [inline test-network/addOrg3/compose/podman/peercfg/core.yaml:523]:
```suggestion
    # Enables/disables the standard out/err from chaincode containers for
    # debugging purposes
    # attachStdout:
    #   false

    # Parameters on creating docker container.
    # Container may be efficiently created using ipam & dns-server for cluster
    # NetworkMode - sets the networking mode for the container. Supported
    # standard values are: `host`(default),`bridge`,`ipvlan`,`none`.
    # Dns - a list of DNS servers for the container to use.
    # Note:  `Privileged` `Binds` `Links` and `PortBindings` properties of
    # Docker Host Config are not supported and will not be used if set.
    # LogConfig - sets the logging driver (Type) and related options
    # (Config) for Docker. For more info,
    # https://docs.docker.com/engine/admin/logging/overview/
    # Note: Set LogConfig using Environment Variables is not supported.
```

File: test-network/compose/docker/peercfg/core.yaml
Comment (satota2) [inline test-network/compose/docker/peercfg/core.yaml:498]:
```suggestion
  # settings for docker vms
```

File: test-network/compose/docker/peercfg/core.yaml
Comment (satota2) [inline test-network/compose/docker/peercfg/core.yaml:524]:
```suggestion
    # Enables/disables the standard out/err from chaincode containers for
    # debugging purposes
    attachStdout:
      false

    # Parameters on creating docker container.
    # Container may be efficiently created using ipam & dns-server for cluster
    # NetworkMode - sets the networking mode for the container. Supported
    # standard values are: `host`(default),`bridge`,`ipvlan`,`none`.
    # Dns - a list of DNS servers for the container to use.
    # Note:  `Privileged` `Binds` `Links` and `PortBindings` properties of
    # Docker Host Config are not supported and will not be used if set.
    # LogConfig - sets the logging driver (Type) and related options
    # (Config) for Docker. For more info,
    # https://docs.docker.com/engine/admin/logging/overview/
    # Note: Set LogConfig using Environment Variables is not supported.
```

File: test-network/compose/podman/peercfg/core.yaml
Comment (satota2) [inline test-network/compose/podman/peercfg/core.yaml:534]:
```suggestion
  # settings for docker vms
  # docker:
  #   tls:
  #     enabled: false
  #     ca:
  #       file: docker/ca.crt
  #     cert:
  #       file: docker/tls.crt
  #     key:
  #       file: docker/tls.key

    # Enables/disables the standard out/err from chaincode containers for
    # debugging purposes
    # attachStdout:
    #   false

    # Parameters on creating docker container.
    # Container may be efficiently created using ipam & dns-server for cluster
    # NetworkMode - sets the networking mode for the container. Supported
    # standard values are: `host`(default),`bridge`,`ipvlan`,`none`.
    # Dns - a list of DNS servers for the container to use.
    # Note:  `Privileged` `Binds` `Links` and `PortBindings` properties of
    # Docker Host Config are not supported and will not be used if set.
    # LogConfig - sets the logging driver (Type) and related options
    # (Config) for Docker. For more info,
    # https://docs.docker.com/engine/admin/logging/overview/
    # Note: Set LogConfig using Environment Variables is not supported.
    # hostConfig:
    #   NetworkMode: host
    #   Dns:
    #     # - 192.168.0.1
    #   LogConfig:
    #     Type: json-file
    #     Config:
    #       max-size: "50m"
    #       max-file: "5"
    #   Memory: 2147483648

```

**Review by satota2 — COMMENTED** [review]
@bestbeforetoday
The updates look good to me. I have corrected the indentation for some of the commented-out configuration values, so please review those adjustments.

**Comment by satota2** [comment]
@bestbeforetoday 
Through this PR, I noticed that the original YAML file in the Fabric core repository also uses 4 spaces for indentation.
https://github.com/hyperledger/fabric/blob/main/sampleconfig/core.yaml

So, I plan to submit a separate PR to the Fabric core repository to change this to 2 spaces.
I believe that aligning the indentation between the sample configuration in Fabric core and the configuration in fabric-samples’ test network will be beneficial for future updates and for users when making comparisons.

**Comment by satota2** [comment]
@bestbeforetoday 

> No problem! I didn't set out to change the whitespace intentionally. I just have my editor set to autoformat on save. If you are happy with the 2 space indentation, the changes are (much) easier to review if you set GitHub's diff view settings to hide whitespace.

Thank you for your comment! I just saw it, and I appreciate the suggestion—I wasn't aware of that option, so it's very helpful. 

> So, I plan to submit a separate PR to the Fabric core repository to change this to 2 spaces.

Using this option, the indentation differences may indeed be less noticeable when comparing with the original file. That said, I think I'll still go ahead and submit a PR to adjust the indentation in the original file, leaving it to the maintainers to decide.

**Comment by bestbeforetoday** [comment]
Thank you for taking the time to go through all the formatting. Unfortunately, applying them all can't be done in one go and they are a pain to work through in the GitHub diff editor - I lost my batch updates several times. I have updated this PR to make the minimal changes required to resolve the actual problem being addressed. Formatting can be tackled in a separate PR when I or somebody else has time.

**Comment by bestbeforetoday** [comment]
Given that these files are very similar to the [sampleconfig](https://github.com/hyperledger/fabric/blob/main/sampleconfig/core.yaml) in the main Fabric repo, but are already slightly out-of-date compared to Fabric's version, I wonder if it is worth removing all the problematic commented out sections and comments. Instead, a comment at the top of each file could reference the Fabric sampleconfig. Entirely up to you though.

**Review by satota2 — APPROVED** [review]
@bestbeforetoday
Understood! I’ve reviewed the newly updated commit, and there are no issues with the changes. Thank you for handling this.

**Comment by satota2** [comment]
> Given that these files are very similar to the [sampleconfig](https://github.com/hyperledger/fabric/blob/main/sampleconfig/core.yaml) in the main Fabric repo, but are already slightly out-of-date compared to Fabric's version, I wonder if it is worth removing all the problematic commented out sections and comments. Instead, a comment at the top of each file could reference the Fabric sampleconfig. Entirely up to you though.

Thank you for your comment. As noted below, I’ve submitted a patch to fix the indentation in the original files:
https://github.com/hyperledger/fabric/pull/5050

Once this is merged, I think it would be a good approach to update the various files in test-network based on the original file, taking your suggestion into consideration as well.

**Pull request merged**
